pkg/controllers/sync/dispatch/retain.go

```diff
@@ -470,6 +470,9 @@ func retainContainers(desiredContainers, clusterContainers []interface{}) error
 }
 
 func retainContainer(desiredContainer, clusterContainer map[string]interface{}) error {
+	resources, _, _ := unstructured.NestedFieldNoCopy(clusterContainer, "resources")
```

Comment:
What if `NestedFieldNoCopy` returns error or !exists? Pls explicitly handle boundary cases.

Reply:
Some cloud vendors, when creating pods, add resource configurations through webhooks, causing continuous update failures during deployment. Therefore, when distributing pods, we retain the spec.containers.resources field.Pls help me check if it is necessary to add it.

Reply:
When I constructed an empty map, the unit test did fail. So I added the necessary check.